89.md

```diff
@@ -0,0 +1,107 @@
+NIP-89
+======
+
+Recommended Application Handlers
+--------------------------------
+
+`draft` `optional` `author:pablof7z`
+
+This NIP describes `kind:31989` and `kind:31990`: a way for unknown event-kinds application-handlers to be discovered.
+
+## Rationale
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics. This NIP seeks to provide the glue for cross-`kind` client interaction.
+
+### Parties involved
+There are three actors to this workflow:
+
+* application that handles a specific event kind
+    * Publishes `kind:31990`, detailing how apps should redirect to it
+* user who recommends an app that handles a specific event kind
+    * Publishes `kind:31989`
+* user who seeks a recommendation for an app that handles a specific event kind
+    * Queries for `kind:31989` and, based on results, queries for `kind:31990`
+
+# Events
+
+## Recommendation event
+```json
+{
+    "kind": 31989,
+    "pubkey": <recommender-user-pubkey>,
+    "tags": [
+        [ "d", <supported-event-kind> ],
+        [ "p", <app-pubkey> ]
+    ]
+}
+```
+
+## Handler information
+```json
+{
+    "kind": 31990,
+    "pubkey": <application-pubkey>,
+    "tags": [
+        [ "d", <supported-event-kind> ]
+        [ "web", "https://example.com/", "root" ],
+        [ "web", "https://..../a/<bech32>", "nevent" ],
+        [ "web", "https://..../p/<bech32>", "nprofile" ],
+        [ "web", "https://..../e/<bech32>" ],
+        [ "ios", ".../<bech32>" ]
+    ]
+}
+```
+
+`bech32` in a URL MUST be replaced by clients with the NIP-19-encoded entity that should be loaded by the application.
+
+Multiple tags might be registered by the app, following NIP-19 nomenclature as the second value of the array.
+
+A special `root` tag MIGHT be included with the main page of the application.
+
+A tag without a second value in the array SHOULD be considered a generic handler for any NIP-19 entity that is not handled by a different tag.
+
+# User flow
+A user A who uses a non-`kind:1`-centric nostr app could choose to announce/recommend a certain kind-handler application.
+
+When user B sees an unknown event kind, e.g. in a social-media centric nostr client, the client would allow user B to interact with the unknown-kind event (e.g. tapping on it).
+
+The client MIGHT query for the user's and the user's follows handler.
+
+# Example
+
+## User A recommends a `kind:31337`-handler
+User A might be a user of Zapstr, a `kind:31337`-centric client (tracks). Using Zapstr, user A publishes an event recommending Zapstr as a `kind:31337`-handler.
+
```

Comment:
One random idea is that maybe signing apps (Alby etc) could have a blacklist of kinds such that if an app asks to use it, they'd ask for user confirmation every single time. At least it would add some friction and would make it obvious to the first users so they would raise the alarms.

Reply:
this is just like nostr: we don't need to solve this in the same way nostr chooses not to solve for DIDs and stuff like that; we can let the free market decide; apps that are abusive will naturally be shunned and clients/users can choose to blacklist the recommendation of apps.

Ultimately, all nostr events are signed by the user's key, which means is the users' responsibility, in the same way that you might give a friend your email address and a virus in their computer spam the hell out of you, it's up to you and the user to figure out, not a concern of SMTP whom/how you should give your email address to 😂